tvm/src/codegen/hlsc/build_hlsc.cc

```diff
@@ -46,6 +46,9 @@ std::string BuildHLSC(
     if (mode == OutputMode::DeviceOnly) {
       map_arg_type["sdaccel"] = 
           std::make_tuple("sdaccel", Handle());
+    } else if (mode == OutputMode::VHLSDevice) {
```

Comment:
Is this map `map_arg_type["vhls"]` used anywhere?

Reply:
No... This is somehow tricky. If I directly set `mode=Output::DeviceOnly`, `map_arg_type["sdaccel"]` is used, and `ptr_mode` of `CodeGenVivadoHLS` is set to true.
https://github.com/cornell-zhang/heterocl/blob/e34f337bdd3350cab1d63e2dd0c38a2125c7867d/tvm/src/codegen/hlsc/codegen_vhls.cc#L47-L50

Then an OpenCL kernel is generated instead of a VHLS kernel.
https://github.com/cornell-zhang/heterocl/blob/e34f337bdd3350cab1d63e2dd0c38a2125c7867d/tvm/src/codegen/hlsc/codegen_vhls.cc#L566-L567

Reply:
If the OpenCL code is returned, that means that the `CodeGenVHLS` in `codegen_vhls.cc` is not invoked. The `map_arg_type` is a flag to direct code generator to generate code targeting different toolflows. For example, SDSoC and SDAceel both use the VHLS CodeGen, but their syntax is kind of different. That's why I introduced a flag to further customize the code generator in different cases.

The logic to determine the CodeGen types. Different code generators are specified for different tool flows (For example, we use CodegenXOCLhost to generate host code for SDAccel or Vitis). I guess you may somehow run into some other conditions listed here:
https://github.com/cornell-zhang/heterocl/blob/e34f337bdd3350cab1d63e2dd0c38a2125c7867d/tvm/src/codegen/build_common.cc#L221-L254

Reply:
Here "sdsoc" and "vivado_hls" use the same codegen.